tensilelite/Tensile/Source/client/source/DataInitialization.cpp

```diff
@@ -1744,8 +1793,13 @@ namespace TensileLite
                     auto tiledSize = desc.sizes()[1];
                     auto tmpTensor = Tensor::create<Half>({tiledSize, unrolledSize});
                     memcpy(tmpTensor.as<void>(), p.cpuInput.valid.get(), tmpTensor.getNumBytes());
-                    tmpTensor.reshape({tiledSize / MiM, MiM, unrolledSize / (MiK * PackK), MiK / MiKv , MiKv * PackK});
-                    Tensor permuted = permute(tmpTensor, {0, 2, 3, 1, 4});
+                    ::Tensor::Manipulation::Shape paddedShape{((tiledSize / MiM) + !!(tiledSize % MiM)) * MiM,
+                        (unrolledSize / (MiK * PackK) + !!(unrolledSize % (MiK * PackK))) * MiK * PackK};
+                    //Temporary hack
+                    uint64_t padVal{};
+                    auto paddedTensor = ::Tensor::Manipulation::pad(tmpTensor, paddedShape, &padVal, tmpTensor.getElementSize());
+                    paddedTensor.reshape({paddedShape[0] / MiM, MiM, paddedShape[1] / (MiK * PackK), MiK / MiKv , MiKv * PackK});
+                    Tensor permuted = permute(paddedTensor, {0, 2, 3, 1, 4});
                     ptr = copyInputBuffers(desc,
```

Comment:
If my understanding is correct, this will use 0 as the padding values (padVal). Have you tried to use random values as the padding values and make sure the validation still pass ?

Reply:
Yes, here I use 0 as padding values, I've tried other value as padding, validation is still passed.